Source/AwsCommonRuntimeKit/http/HTTPClientConnectionOptions.swift

```diff
@@ -42,6 +43,47 @@ public struct HTTPClientConnectionOptions: CStructWithShutdownOptions {
 
     public var monitoringOptions: HTTPMonitoringOptions?
 
+    /// (Optional) HTTP/2 specific configuration
+    /// Specify whether you have prior knowledge that cleartext (HTTP) connections are HTTP/2 (RFC-7540 3.4).
+    /// If false, then cleartext connections are treated as HTTP/1.1.
+    /// It is illegal to set this true when secure connections are being used.
+    /// Note that upgrading from HTTP/1.1 to HTTP/2 is not supported (RFC-7540 3.2).
+    public var http2PriorKnowledge: Bool
```

Comment:
ask dengke if we need to add these options to the vanilla connection manager. I'm not sure anyone would use this to create HTTP/2 connections. I think we added these in C so the stream manager could use the connection manager under the hood.

if you can just leave these out, go for it. for all we know we'll be making yet-another connection manager that handles both HTTP/1 and 2 in the next few months